drivers/r2dbc-driver/src/main/kotlin/app/cash/sqldelight/driver/r2dbc/R2dbcDriver.kt

```diff
@@ -26,10 +28,9 @@ class R2dbcDriver(private val connection: Connection) : SqlDriver {
     return QueryResult.AsyncValue {
       val result = prepared.execute().awaitSingle()
 
-      val rowSet = mutableListOf<Map<Int, Any?>>()
-      result.map { row, rowMetadata ->
-        rowSet.add(rowMetadata.columnMetadatas.mapIndexed { index, _ -> index to row.get(index) }.toMap())
-      }.awaitLast()
```

Comment:
awaitLast requires at least one item, but this does not work if your result is empty, eg because the query does not return any rows.